src/yoroi-wallets/cardano/byron/ByronWallet.ts

```diff
@@ -993,19 +988,19 @@ export class CardanoWallet implements YoroiWallet {
   }
 
   // TODO: caching
-  async fetchNfts(): Promise<YoroiNft[]> {
+  fetchNfts(): Promise<YoroiNft[]> {
     const utxos = this.utxos
     const assets = utxos.flatMap((utxo) => utxo.assets ?? [])
 
     if (assets.length === 0) {
-      return []
+      return Promise.resolve([])
     }
 
     return api.getNFTs(assets, this.getBackendConfig())
   }
 
   // TODO: caching
-  async fetchNftModerationStatus(fingerprint: string): Promise<YoroiNftModerationStatus> {
+  fetchNftModerationStatus(fingerprint: string): Promise<YoroiNftModerationStatus> {
     const backendConfig = this.getBackendConfig()
     const networkConfig = this.getNetworkConfig()
     const isMainnet = networkConfig.IS_MAINNET
```

Comment:
some of these checks should go away but maybe we keep the Api, as it is until we inject it in wallet